UI: Bugfix - provision volumes with labels

carlito-scality opened the pull request:
**Component**: UI 

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
- We need to add labels in the templates instead of the Volume object itself when creating volumes so labels can be propagated to PersistentVolume.

**Summary**:

**Acceptance criteria**: 

<img width="741" alt="Screenshot 2019-11-22 at 10 08 17" src="https://user-images.githubusercontent.com/47394132/69413619-75316d00-0d11-11ea-8484-b1f219223844.png">

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/1891

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
UI: Provision volumes with labels - Pass the labels in the template of Volume when creating so labels can be set correctly in PersistentVolume.
- Display the peristentVolume labels instead of Volume labels ifself

carlito-scality committed:
UI: make labels visible on Dark Mode when creating volumes

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/provision_volumes_with_labels` into
`development/2.4`
* [w/2.5/bugfix/provision_volumes_with_labels](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/provision_volumes_with_labels) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

ChengYanJin reviewed (dismissed):
LGTM

gdemonet commented on ui/src/ducks/app/volumes.js:
```diff
@@ -212,11 +212,15 @@ export function* createVolumes({ payload }) {
     kind: 'Volume',
     metadata: {
       name: newVolume.name,
-      labels: newVolume.labels,
```
No, keep it

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #94bed350fee2418e8e0c3118a630a0d7452f82f1 on the integration branch
`w/2.5/bugfix/provision_volumes_with_labels` is merging a branch which is neither the current
branch `bugfix/provision_volumes_with_labels` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/provision_volumes_with_labels` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

carlito-scality committed:
Cypress tests: update volume creation tests
- Add cypress-wait-until library to check the volume avaibility by retrying the status check
- Fix Volume SparseLoopDevice failed status due to the size problem

carlito-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/provision_volumes_with_labels` into
`development/2.4`
* [w/2.5/bugfix/provision_volumes_with_labels](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/provision_volumes_with_labels) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #aa709d0669bec793ebafc64eda377e2180950545 on the integration branch
`w/2.5/bugfix/provision_volumes_with_labels` is merging a branch which is neither the current
branch `bugfix/provision_volumes_with_labels` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/provision_volumes_with_labels` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

carlito-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/provision_volumes_with_labels` into
`development/2.4`
* [w/2.5/bugfix/provision_volumes_with_labels](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/provision_volumes_with_labels) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

carlito-scality committed:
UI: install cypress-wait-until in cypress.sh
- cypress-wait-until is required to launch cypress tests

carlito-scality commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/209b2d60602764705e121089c97450e2e68f9624) did not succeed in branch w/2.5/bugfix/provision_volumes_with_labels.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye carlito-scality.
